addons/mail_group/models/mail_group_message.py

```diff
@@ -0,0 +1,271 @@
+        required=True, default='pending_moderation')
+    moderator_id = fields.Many2one('res.users', string='Moderated By')
+
+    @api.depends('email_from')
+    def _compute_email_from_normalized(self):
+        for message in self:
+            message.email_from_normalized = email_normalize(message.email_from)
+
+    @api.depends('mail_message_id.parent_id')
+    def _compute_group_parent_id(self):
+        # Perform only one search for all messages
+        all_parents = self.mapped('mail_message_id.parent_id')
+        if all_parents:
+            parent_per_message = {
+                message.mail_message_id: message
+                for message in self.search([('mail_message_id', 'in', all_parents.ids)])
+            }
+        else:
+            parent_per_message = {}
+
+        for message in self:
+            message.group_parent_id = parent_per_message.get(message.mail_message_id.parent_id)
+
+    def _inverse_group_parent_id(self):
+        for message in self:
+            message.mail_message_id.parent_id = message.group_parent_id.mail_message_id
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('mail_message_id'):
+            vals['res_id'] = vals.get('mail_group_id')
+            vals['model'] = 'mail.group'
+            vals['mail_message_id'] = self.env['mail.message'].create({
+                field: vals.pop(field)
+                for field in self.env['mail.message']._fields
+                if field in vals
+            }).id
+        return super(MailGroupMessage, self).create(vals)
+
+    def copy(self, default=None):
+        default = dict(default or {})
+        default['mail_message_id'] = self.mail_message_id.copy().id
+        return super(MailGroupMessage, self).copy(default)
+
+    # --------------------------------------------------
+    # MODERATION API
+    # --------------------------------------------------
+
+    def action_accept(self):
+        """Accept the incoming email.
+
+        Will send the incoming email to all members of the group.
+        """
+        self._assert_moderable()
+        self.write({
+            'moderation_status': 'accepted',
+            'moderator_id': self.env.uid,
+        })
+
+        # Send the email to the members of the group
+        alias_domain = self.env['ir.config_parameter'].sudo().get_param('mail.catchall.domain')
+        for message in self:
+            mail_group = message.mail_group_id
+
+            reply_to = None
+            if mail_group.alias_name and alias_domain:
+                reply_to = f'{mail_group.alias_name}@{alias_domain}'
+            else:
+                _logger.error('The alias or the catchall domain is missing, group might not work properly.')
+
+            base_url = self.env['mail.group'].get_base_url()
+            body = self.env['mail.render.mixin']._replace_local_links(message.body)
+            mail_values = []
+            for member in mail_group.member_ids:
```

Comment:
I think notification should be done in its own method on group model, so that it may be explicitely called. Indeed acceptance is a moderation process while sending emails is related to group. Accepting a message should simply call the notification method on group with given message_id as parameter. What do you think ?

Reply:
Hum, ok, so we "mimic" how `mail.thread` works (and we are coherent)